R/combineFeatures.R

```diff
@@ -320,5 +318,10 @@ robust_summary <- function(e, nIter = 100, residuals = FALSE, ...) {
     ## resids <- fit$residuals
     ## se <- unique(summary(fit)$coefficients[sampleid, 'Std. Error'])
     ## was X[, sampleid, drop = FALSE] %*% fit$coefficients[sampleid]
-    return(fit$coefficients[sampleid])
+
+    ## Put NA for the samples without any expression value
+    present = apply(e,2,function(x){any(!is.na(x))})
+    out = rep(NA,length(present))
+    out[present] = fit$coefficients[sampleid]
+    return(out)
```

Comment:
`return` is just needed if you want to jump out of a `function` (e.g. in an `if`-statement). If you use it at the end it is just an unnecessary function call. That means `out` would be enough.